admin_guide/taints_tolerations.adoc

```diff
@@ -0,0 +1,329 @@
+the special nodes and hence the admission controller should add the toleration.
+
+To make sure pods can only use the specialized hardware, you will need some
+additional mechanism. For example, you could represent the special resource 
+using link:https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#opaque-integer-resources-alpha-feature[opaque integer resources] 
+and request it as a resource in the PodSpec, or you could label the nodes that have
+the special hardware and use node affinity on the pods that need the hardware.
+
+
+## Per-pod-configurable eviction behavior when there are node problems (alpha feature)
+
+The `NoExecute` taint effect, affects pods that are already
+running on the node as follows:
+
+ * pods that do not tolerate the taint are evicted immediately
+ * pods that tolerate the taint without specifying `tolerationSeconds` in
+   their toleration specification remain bound forever
+ * pods that tolerate the taint with a specified `tolerationSeconds` remain
+   bound for the specified amount of time
+
+The above behavior is a beta feature. In addition, Kubernetes 1.6 has alpha
+support for representing node problems (currently only "node unreachable" and
+"node not ready", corresponding to the NodeCondition "Ready" being "Unknown" or
+"False" respectively) as taints. 
+
+When the `TaintBasedEvictions` alpha feature
+is enabled (you can do this by including `TaintBasedEvictions=true` in `--feature-gates`, such as
+`--feature-gates=FooBar=true,TaintBasedEvictions=true`), the taints are automatically
+added by the NodeController and the normal logic for evicting pods from nodes
+based on the Ready NodeCondition is disabled.
+
+[NOTE]
+==== 
+To maintain the existing link:https://kubernetes.io/docs/admin/node/#node-controller[rate limiting]
+behavior of pod evictions due to node problems, the system adds the taints
+in a rate-limited way. This prevents massive pod evictions in scenarios such
+as the master becoming partitioned from the nodes.
+====
+
+This alpha feature, in combination with `tolerationSeconds`, allows a pod
+to specify how long it should stay bound to a node that has one or both of these problems.
+
+For example, you might want an application with a lot of local state to stay
+bound to node for a long time in the event of network partition, in the event
+that the partition recovers and thus the pod eviction can be avoided.
+The toleration the pod would use in that case would look like
+
+----
+yaml
+tolerations: 
+- key: "node.alpha.kubernetes.io/unreachable"
+  operator: "Exists"
+  effect: "NoExecute"
+  tolerationSeconds: 6000
+----
+
+For the node not ready case, change the key to `node.alpha.kubernetes.io/notReady`:
+
+----
+yaml
+tolerations: 
+- key: "node.alpha.kubernetes.io/notReady"
+  operator: "Exists"
+  effect: "NoExecute"
+  tolerationSeconds: 6000
+----
+
+
+Note that {product-title} automatically adds a toleration for
```

Comment:
Added: "For information on the Kubernetes DefaultTolerationSeconds admissions controller plugin that sets these defaults, see the
 link:https://kubernetes.io/docs/admin/admission-controllers/#defaulttolerationseconds[Kubernetes documenation]."